File: Part 2 - Regresion/1 - Simple Linear Regression/simple_linear_regresion.R
# Simple Linear regresion

# Import dataset
dataset = read.csv('Salary_Data.csv')
# Taking care of missing data\


# install.packages('caTools')

# Spliting the datasetset into Training set and Test set
library(caTools)
set.seed(123)
# Split ratio is for training set in this case not for test
# The column used is y
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split==TRUE)
test_set = subset(dataset, split==FALSE)


# Feature scaling
# Not needed because the package does this for us
# training_set[,2:3] = scale(training_set[,2:3])
# test_set[,2:3] = scale(test_set[,2:3])

regressor = lm(formula= Salary ~ YearsExperience, 
               data = training_set)

# Predict the test set results
y_pred = predict(regressor, newdata =  test_set)



# install.packages("ggplot2", dependencies=TRUE)
library(ggplot2)
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red' ) +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
            colour = 'blue') +
  ggtitle('Salary vs Years of experience (Training)') +
  xlab('Years of experience') +
  ylab('Salary')

# Visualizing the test set
ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red' ) +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
            colour = 'blue') +
  ggtitle('Salary vs Years of experience (Test)') +
  xlab('Years of experience') +
  ylab('Salary')

File: Part 2 - Regresion/1 - Simple Linear Regression/simple_linear_regresion.py
# Data preprocessing template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# import dataset
dataset = pd.read_csv("Salary_Data.csv")
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

# Spliting the dataset into train and test set
from sklearn.model_selection import train_test_split
# random_state is the seed
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=1/3, random_state=0)

# Feature scaling
# from sklearn.preprocessing import StandardScaler
# standard_scaler_x = StandardScaler()
# standard_scaler_x.fit(x)
# x_train = standard_scaler_x.transform(x_train)
# x_test = standard_scaler_x.transform(x_test)

# Fitting Simple linear regresor to the training set
from sklearn.linear_model import LinearRegression
regresor = LinearRegression()
model = regresor.fit(x_train, y_train)

# Prediction of the test results
y_pred = regresor.predict(x_test)

# Visualizing the training set results
plt.scatter(x_train, y_train, color='red')
plt.plot(x_train, regresor.predict(x_train), color='blue')
plt.title("Experience VS Salary (Training)")
plt.xlabel('Years of Experience')
plt.ylabel("Salary")
plt.show()

# Visualizing the training set results
plt.scatter(x_test, y_test, color='red')
plt.plot(x_train, regresor.predict(x_train), color='blue') 
# Como es linear regresion y el mismo modelo no es necesario cambiar
plt.title("Experience VS Salary (Test)")
plt.xlabel('Years of Experience')
plt.ylabel("Salary")
plt.show()